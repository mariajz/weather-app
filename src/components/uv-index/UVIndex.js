import React from 'react';
import { StyledText } from '../../commons/styles';
import { CustomIcon, GradientColorBar } from '../../commons/visual-elements';
import { getUVWarnings } from '../../utils/helpers';
import { TitleWrapper, UVIndexWrapper } from './UVIndex.style';

const UVIndex = ({ uvIndexProps }) => {
    const { uvIndex } = uvIndexProps;
    const { type, warning } = getUVWarnings(uvIndex);
    return (
        <>
            <UVIndexWrapper>
                <TitleWrapper>
                    <CustomIcon size={15} iconName="sun" />
                    <StyledText marginLeft={5} size={14}>
                        UV INDEX
                    </StyledText>
                </TitleWrapper>
                <StyledText size={30}>{uvIndex}</StyledText>
                <StyledText size={20}>{type}</StyledText>
                <GradientColorBar
                    sliderPosition={uvIndex}
                    minValue={0}
                    maxValue={12}
                />
                <StyledText size={10}>{warning}</StyledText>
            </UVIndexWrapper>
        </>
    );
};

export default UVIndex;
