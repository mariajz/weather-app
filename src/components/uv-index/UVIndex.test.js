import { render } from '@testing-library/react-native';
import React from 'react';
import { View as MockView } from 'react-native';
import UVIndex from './UVIndex';

const defaultProps = {
    uvIndexProps: {
        uvIndex: 5,
    },
};

jest.mock('../../commons/visual-elements', () => ({
    CustomIcon: () => <MockView />,
    Theme: {
        BackgroundWhite: jest.fn(
            opacity => `mocked rgba(255, 255, 255, ${opacity})`,
        ),
    },
    GradientColorBar: () => <MockView />,
}));

describe('UVIndex', () => {
    it('should render UVIndex', () => {
        const container = render(<UVIndex {...defaultProps} />);

        expect(container).toMatchSnapshot();
    });
});
