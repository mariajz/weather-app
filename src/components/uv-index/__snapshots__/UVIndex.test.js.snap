// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UVIndex should render UVIndex 1`] = `
<View
  style={
    {
      "backgroundColor": "mocked rgba(255, 255, 255, 0.2)",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "height": 170,
      "justifyContent": "space-between",
      "marginTop": 15,
      "paddingBottom": 20,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
      "width": "50%",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View />
    <Text
      marginLeft={5}
      size={14}
      style={
        {
          "color": "white",
          "fontSize": 14,
          "marginLeft": 5,
        }
      }
    >
      UV INDEX
    </Text>
  </View>
  <Text
    size={30}
    style={
      {
        "color": "white",
        "fontSize": 30,
        "marginLeft": 0,
      }
    }
  >
    5
  </Text>
  <Text
    size={20}
    style={
      {
        "color": "white",
        "fontSize": 20,
        "marginLeft": 0,
      }
    }
  >
    Moderate
  </Text>
  <View />
  <Text
    size={10}
    style={
      {
        "color": "white",
        "fontSize": 10,
        "marginLeft": 0,
      }
    }
  >
    Some risk of harm. Wear sunglasses and use sunscreen
  </Text>
</View>
`;
